src/backend/bacon-video-widget-gst-missing-plugins.c

```diff
@@ -473,29 +473,92 @@ missing_codecs_supported (const char **codecs)
+
+static CodecMissingSuggestedLaunchAction *
+codec_missing_suggested_launch_action_new (const gchar *message_text,
+					   const gchar *button_text,
+					   GDesktopAppInfo *desktop_info,
+					   const gchar *mrl)
+{
+	CodecMissingSuggestedLaunchAction *action = g_slice_new0 (CodecMissingSuggestedLaunchAction);
+	action->message_text = message_text;
+	action->button_text = button_text;
+	action->desktop_info = desktop_info;
+	action->mrl = mrl;
+
+	return action;
+}
+
+static void
+codec_missing_suggested_launch_action_free (CodecMissingSuggestedLaunchAction *action)
+{
+	g_object_unref (action->desktop_info);
+	g_slice_free (CodecMissingSuggestedLaunchAction, action);
+}
+
+static CodecMissingSuggestedLaunchAction *
+determine_vlc_launch_action_for_missing_codecs (const gchar *mrl) {
+	GDesktopAppInfo *candidate;
+	const gchar *message_text;
+
+	/* First try org.videolan.VLC.desktop, and if it exists
+	 * use that, otherwise use eos-vlc.desktop */
+	candidate = g_desktop_app_info_new ("org.videolan.VLC.desktop");
+	if (candidate) {
+		message_text = _("This file type is currently unsupported by the default "
+		                 "video player. However, you can use VLC media player "
+		                 "to play it.");
+		return codec_missing_suggested_launch_action_new (message_text,
+								  _("Open with VLC"),
+								  candidate,
+								  mrl);
+	}
+
+	candidate = g_desktop_app_info_new ("eos-vlc.desktop");
+	if (candidate) {
+		message_text = _("This file type is currently unsupported by the default "
+		                 "video player. However, you can use VLC media player "
+		                 "from the App Center to play it.");
+		return codec_missing_suggested_launch_action_new (message_text,
+								  _("Get VLC"),
+								  candidate,
+								  mrl);
+	}
+
+	return NULL;
+}
+
 static void
 codec_confirmation_dialog_response_cb (GtkDialog       *dialog,
                                        GtkResponseType  response_type,
                                        gpointer	 user_data)
 {
 	GError *error = NULL;
-	const char * const *languages = g_get_language_names ();
-	const char *shop_uri = NULL;
+	CodecMissingSuggestedLaunchAction *action = (CodecMissingSuggestedLaunchAction *) user_data;
 
 	switch (response_type) {
 	case GTK_RESPONSE_ACCEPT:
-		/* Different URLs for the three different shops */
-		if (g_strv_contains (languages, "es"))
-			shop_uri = "https://shop.endlessm.com/products/codecs";
-		else if (g_strv_contains (languages, "pt"))
-			shop_uri = "https://comprar.endlessm.com/products/codecs";
-		else
-			shop_uri = "https://endless-global.myshopify.com/products/codecs";
+		if (action) {
+			GList *uris = g_list_append (NULL, (gpointer) action->mrl);
+			if (!g_app_info_launch_uris (G_APP_INFO (action->desktop_info),
+						     uris,
+						     NULL,
+						     &error)) {
+				g_warning ("Could not launch %s: %s",
+					   g_desktop_app_info_get_filename (action->desktop_info),
```

Comment:
I'd use `g_app_info_get_id()` here, rather than the filename of the desktop file.